src/Quadrature.cpp

```diff
@@ -0,0 +1,1305 @@
+    for (int p=0; p < _num_polar/2; p++) {
+      if (weights[ap] < 0. || weights[ap] > M_PI_2)
+        log_printf(ERROR, "Unable to polar weight to %f which is "
+                   "not in the range [0,PI/2]", weights[ap]);
+
+      setPolarValues(_polar_weights, a, p, weights[ap]);
+      ap++;
+    }
+  }
+}
+
+
+/**
+ * @brief Sets the polar angle for the given indexes
+ * @param theta the value of the polar angle to be set
+ * @param azim the azimuthal index of the angle of interest
+ * @param polar the polar index of the angle of interest
+ */
+void Quadrature::setTheta(double theta, int azim, int polar) {
+
+  if (theta <= 0.0 || theta >= M_PI_2)
+    log_printf(ERROR, "Unable to set theta for azim = %d and polar = %d "
+               "to %f which is not in the range (0.0, PI/2)", azim, polar,
+               theta);
+
+  else if (azim >= _num_azim/4)
+    log_printf(ERROR, "Unable to set theta for azim = %d and polar = %d "
+               "since azim is not in the range (0, _num_azim/4)", azim, polar);
+
+  else if (polar >= _num_polar/2)
+    log_printf(ERROR, "Unable to set theta for azim = %d and polar = %d "
+               "since polar is not in the range (0, _num_polar/2)",
+               azim, polar);
+
+
+  if (_thetas == NULL) {
+    _thetas = new double*[_num_azim/2];
+    for (int i=0; i < _num_azim/2; i++)
+      _thetas[i] = new double[_num_polar];
+  }
+
+  _thetas[azim][polar] = theta;
+  _thetas[_num_azim/2 - azim - 1][polar] = theta;
+  _thetas[azim][_num_polar - polar - 1] = M_PI - theta;
+  _thetas[_num_azim/2 - azim - 1][_num_polar - polar - 1] = M_PI - theta;
+}
+
+
+/**
+ * @brief Sets the azimuthal angle for the given index
+ * @param phi the value of the azimuthal angle to be set
+ * @param azim the azimuthal index
+ */
+void Quadrature::setPhi(double phi, int azim) {
+
+  if (phi <= 0.0 || phi >= M_PI_2)
+    log_printf(ERROR, "Unable to set phi for azim = %d to %f which is not "
+               "in the range (0.0, PI/2)", azim, phi);
+
+  else if (azim >= _num_azim/4)
+    log_printf(ERROR, "Unable to set phi for azim = %d since azim is not in"
+               " the range (0, _num_azim/4)", azim);
+
+  else if (_phis == NULL)
+    _phis = new double[_num_azim/2];
+
+  _phis[azim] = phi;
+  _phis[_num_azim/2 - azim - 1] = M_PI - phi;
+}
+
+
+/**
+ * @brief Sets the azimuthal spacing for the given index
+ * @param spacing the spacing in the azimuthal direction to be set
+ * @param azim the azimuthal index
+ */
+void Quadrature::setAzimSpacing(FP_PRECISION spacing, int azim) {
+
+  if (spacing <= 0.0)
+    log_printf(ERROR, "Unable to set azimuthal spacing for azim = %d to %f "
+                      "which is not strictly greater than zero", azim,
+                      spacing);
+
+  else if (azim >= _num_azim/4)
+    log_printf(ERROR, "Unable to set azimuthal spacing for azim = %d since "
+                      " azim is not in the range (0, _num_azim/4)", azim);
+
+  else if (_azim_spacings == NULL)
```

Comment:
Although it doesn't actually matter here, I think it is better to make this memory allocation block a standalone `if` rather than an `else if` since it doesn't logically connect to the error checking blocks above.